src/test/java/chess/service/ChessServiceTest.java

```diff
@@ -0,0 +1,103 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import chess.dao.ChessGameDao;
+import chess.dao.PieceDao;
+import chess.domain.Command;
+import chess.dto.ChessGameDto;
+
+@SpringBootTest
+@Transactional
+class ChessServiceTest {
+    private ChessService chessService;
+    private int savedId;
+
+    @Autowired
+    private ChessGameDao chessGameDao;
+    @Autowired
+    private PieceDao pieceDao;
+
+    @BeforeEach
+    void setUp() {
+        chessService = new ChessService(chessGameDao, pieceDao);
+        savedId = chessService.save("test_game1", "1234");
+        chessService.save("test_game2", "1234");
+        chessService.save("test_game3", "1234");
+    }
+
+    @Test
+    @DisplayName("게임을 한개 더 생성하면 4개가 되어야 합니다.")
+    void testGamesSize() {
+        chessService.save("test_game4", "1234");
+        List<ChessGameDto> games = chessService.findAllChessGames();
+        assertThat(games).hasSize(4);
+    }
+
+    @Test
+    @DisplayName("게임을 생성하면 말의 개수가 32개여야 합니다.")
+    void testPiecesOfGames() {
+        int savedId = chessService.save("test_game4", "1234");
+        List<String> chessBoard = chessService.findChessBoardById(savedId);
+        List<String> existPieces = chessBoard.stream()
+            .filter(piece -> !piece.equals(""))
+            .collect(Collectors.toList());
+
+        assertThat(existPieces).hasSize(32);
+    }
+
+    @Test
+    @DisplayName("기물을 정상적으로 움직일 수 있어야 합니다.")
+    void move() {
+        chessService.start(savedId);
+        assertDoesNotThrow(() -> chessService.move("a2", "a4", savedId));
+    }
+
+    @Test
+    @DisplayName("진행중인 게임을 삭제하면 예외를 반환해야 합니다.")
+    void deleteNotEndGame() {
+        assertThatThrownBy(() -> chessService.delete("1234", savedId))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    @DisplayName("게임을 삭제할 때 비밀번호가 틀리면 삭제하지 않아야 합니다.")
+    void deleteWrongPassword() {
+        chessService.finish(Command.from("end"), savedId);
+        chessService.delete("123", savedId);
+        List<String> chessBoard = chessService.findChessBoardById(savedId);
+        assertThat(chessBoard).isNotNull();
+    }
+
+    @Test
+    @DisplayName("끝난 게임을 정확한 비밀번호로 삭제할 시 오류가 발생하지 않아야 합니다.")
+    void delete() {
+        chessService.finish(Command.from("end"), savedId);
+        assertDoesNotThrow(() -> chessService.delete("1234", savedId));
+    }
+
+    @Test
+    @DisplayName("게임을 종료하면 게임의 상태는 종료상태여야 합니다.")
+    void end() {
+        chessService.finish(Command.from("end"), savedId);
+        assertThat(chessService.isEnd(savedId)).isTrue();
+    }
+
+    @Test
+    @DisplayName("게임을 종료하면 이긴 팀의 이름을 반환해야 합니다.")
+    void finish() {
+        String teamName = chessService.finish(Command.from("end"), savedId);
+        assertThat(teamName).isEqualTo("DRAW");
+    }
+}
```

Comment:
```suggestion
}

```

EOF를 챙겨주세요~

- 여러 버전의 유닉스 OS간의 공통된 API 및 인터페이스를 정의해놓은 POSIX 명세 중 text file 부분을 보면 유닉스 시스템은 text file들을 서로 구분할때 EOL을 사용하기 때문에 파일을 만들때마다 파일의 끝에 EOL을 넣어주는 것이 안전하다.

Reply:
이거 딴 사람꺼는 잘보이는데 제껀 못봤네요..ㅋㅋ